modules/test/playwright/tests/api-builder/apiFilter.spec.ts

```diff
@@ -0,0 +1,131 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {APIResponse, expect as baseExpect, mergeTests} from '@playwright/test';
+
+import {apiHelpersTest} from '../../fixtures/apiHelpersTest';
+import {loginTest} from '../../fixtures/loginTest';
+
+export const test = mergeTests(apiHelpersTest, loginTest);
+
+const expect = baseExpect.extend({
+	toBeSuccessful: (response: APIResponse) => ({
+		message: () =>
+			response.ok()
+				? 'Response is successful'
+				: 'Response is not successful',
+		pass: response.ok(),
+	}),
+	toHaveItems: async (response: APIResponse, expectedItems: object[]) => {
+		baseExpect((await response.json()).items).toEqual(expectedItems);
+
+		return {
+			message: () => 'The items are correct',
+			pass: true,
+		};
+	},
+});
+
+test('can GET with API Filter', async ({apiHelpers}) => {
+	await apiHelpers.featureFlag.updateFeatureFlag('LPS-178642', true);
+
+	const objectDefinition =
+		await apiHelpers.objectAdmin.postRandomObjectDefinition('');
+
+	const apiApplication = await apiHelpers.apiBuilder.postApiApplication({
+		apiApplicationToAPIEndpoints: [
+			{
+				description: 'description',
+				externalReferenceCode: 'ENDPOINT',
+				httpMethod: 'get',
+				name: 'GetEndpoint',
+				path: '/path-endpoint',
+				scope: 'company',
+			},
+		],
+		apiApplicationToAPISchemas: [
+			{
+				apiSchemaToAPIProperties: [
+					{
+						description: 'description',
+						name: 'textProperty',
+						objectFieldERC: 'textField',
+					},
+				],
+				description: 'description',
+				externalReferenceCode: 'SCHEMA',
+				mainObjectDefinitionERC: objectDefinition.externalReferenceCode,
+				name: 'schema',
+			},
+		],
+		applicationStatus: 'published',
+		baseURL: 'test',
+		externalReferenceCode: 'APPLICATION',
+		title: 'title',
+	});
+
+	expect(apiApplication.status).toStrictEqual({
+		code: 0,
+		label: 'approved',
+		label_i18n: 'Approved',
+	});
+
+	expect(
+		await apiHelpers.apiBuilder.putResponse(
+			'schemas/by-external-reference-code/SCHEMA/requestAPISchemaToAPIEndpoints/ENDPOINT'
```

Comment:
You should parametrize `ENDPOINT` with what you already have in the `apiApplication` json. That way if the object definition changes in the future we don't need to change the hardcoded values.

Reply:
this is left on purpose so the test is more readable :)